app/lib/dartdoc/dartdoc_runner.dart

```diff
@@ -19,7 +28,68 @@ class DartdocRunner implements TaskRunner {
 
   @override
   Future<bool> runTask(Task task) async {
-    // TODO: implement doc generation and upload
-    return false; // no race detected
+    final tempDir =
+        await Directory.systemTemp.createTemp('pub-dartlang-dartdoc');
+    final pubCacheDir = await tempDir.resolveSymbolicLinks();
+
+    final pubEnv = new PubEnvironment(pubCacheDir: pubCacheDir);
+
+    try {
+      final pkgLocation =
+          await pubEnv.getLocation(task.package, version: task.version);
+      final pkgPath = pkgLocation.location;
+      await _runDartdoc(task, pkgPath);
+      await _writeMetadata(task, pkgPath);
+      // TODO: upload doc/api to the appropriate bucket
+    } finally {
+      await tempDir.delete(recursive: true);
+    }
+    return false; // no race detection
+  }
+
+  Future _runDartdoc(Task task, String pkgPath) async {
+    final pr = await Process.run('dartdoc', [], workingDirectory: pkgPath);
+    if (pr.exitCode != 0) {
+      _logger.severe('Error while running dartdoc for $task.\n'
+          'exitCode: ${pr.exitCode}\n'
+          'stdout: ${pr.stdout}\n'
+          'stderr: ${pr.stderr}\n');
+    }
```

Comment:
Please throw an exception or give a boolean to the caller to indicate whether it was successful. If we fail to generate dartdocs, we should not try running gcs uploading code!